app/src/main/java/org/oppia/android/app/mydownloads/MyDownloadsActivity.kt

```diff
@@ -17,20 +20,29 @@ class MyDownloadsActivity : InjectableAppCompatActivity() {
   override fun onCreate(savedInstanceState: Bundle?) {
     super.onCreate(savedInstanceState)
     activityComponent.inject(this)
-    myDownloadsActivityPresenter.handleOnCreate()
+    val isFromNavigationDrawer = intent.getBooleanExtra(
+      BOOL_IS_FROM_NAVIGATION_DRAWER_EXTRA_KEY,
+      /* defaultValue= */ false
+    )
     internalProfileId = intent.getIntExtra(KEY_NAVIGATION_PROFILE_ID, -1)
+    myDownloadsActivityPresenter.handleOnCreate(internalProfileId, isFromNavigationDrawer)
   }
 
   companion object {
-    fun createMyDownloadsActivityIntent(context: Context, profileId: Int?): Intent {
+
+    internal const val BOOL_IS_FROM_NAVIGATION_DRAWER_EXTRA_KEY =
+      "BOOL_IS_FROM_NAVIGATION_DRAWER_EXTRA_KEY"
+
+    fun createMyDownloadsActivityIntent(
+      context: Context,
+      internalProfileId: Int?,
+      isFromNavigationDrawer: Boolean
```

Comment:
I think we can remove it and if in future we need this functionality we can introduce it back.
For now `MyDownloads` will start from `NavigationDrawer` only.

WDYT?

Reply:
Keeping it open, till I sent https://github.com/oppia/oppia-android/pull/3072 for review